crates/ide/src/references.rs

```diff
@@ -1166,18 +1166,67 @@ fn foo<const FOO$0: usize>() -> usize {
     fn test_find_self_ty_in_trait_def() {
         check(
             r#"
-trait Foo {
+trait Foo where Self: {
     fn f() -> Self$0;
 }
 "#,
             expect![[r#"
                 Self TypeParam FileId(0) 6..9 6..9
 
-                FileId(0) 26..30
+                FileId(0) 16..20
+                FileId(0) 38..42
             "#]],
         );
+        //         check(
+        //             r#"
+        // trait Foo$0 where Self: {
+        //     fn f() -> Self;
+        // }
+        // "#,
+        //             expect![[r#"
+        //                 Foo Trait FileId(0) 0..45 6..9
+
+        //                 FileId(0) 16..20
+        //                 FileId(0) 38..42
+        //             "#]],
+        //         );
```

Comment:
Are we somehow able to go from a `hir::Trait` to its corresponding SelfType?

Reply:
We probably shouldn’t — in traits, Self is a type parameter. That is, if you want flag Self as usage of Foo, then you should flag T in `T: Foo` as well.

Reply:
Right, I didn't think of that